lib/views/issue-detail-view.js

```diff
@@ -0,0 +1,246 @@
+import PropTypes from 'prop-types';
+import cx from 'classnames';
+
+import IssueTimelineController from '../controllers/issue-timeline-controller';
+import Octicon from '../atom/octicon';
+import IssueishBadge from '../views/issueish-badge';
+import GithubDotcomMarkdown from '../views/github-dotcom-markdown';
+import EmojiReactionsView from '../views/emoji-reactions-view';
+import PeriodicRefresher from '../periodic-refresher';
+import {autobind} from '../helpers';
+import {addEvent} from '../reporter-proxy';
+
+export class BareIssueDetailView extends React.Component {
+  static propTypes = {
+    relay: PropTypes.shape({
+      refetch: PropTypes.func.isRequired,
+    }),
+    switchToIssueish: PropTypes.func.isRequired,
+    repository: PropTypes.shape({
+      id: PropTypes.string.isRequired,
+      name: PropTypes.string.isRequired,
+      owner: PropTypes.shape({
+        login: PropTypes.string,
+      }),
+    }),
+    issueish: PropTypes.shape({
+      __typename: PropTypes.string.isRequired,
+      id: PropTypes.string.isRequired,
+      title: PropTypes.string,
+      url: PropTypes.string.isRequired,
+      bodyHTML: PropTypes.string,
+      number: PropTypes.number,
+      state: PropTypes.oneOf([
+        'OPEN', 'CLOSED',
+      ]).isRequired,
+      author: PropTypes.shape({
+        login: PropTypes.string.isRequired,
+        avatarUrl: PropTypes.string.isRequired,
+        url: PropTypes.string.isRequired,
+      }).isRequired,
+      reactionGroups: PropTypes.arrayOf(
+        PropTypes.shape({
+          content: PropTypes.string.isRequired,
+          users: PropTypes.shape({
+            totalCount: PropTypes.number.isRequired,
+          }).isRequired,
+        }),
+      ).isRequired,
+    }).isRequired,
+  }
+
+  state = {
+    refreshing: false,
+  }
+
+  constructor(props) {
+    super(props);
+    autobind(this, 'handleRefreshClick', 'refresh', 'renderIssueBody');
+  }
+
+  componentDidMount() {
+    this.refresher = new PeriodicRefresher(BareIssueDetailView, {
+      interval: () => 5 * 60 * 1000,
+      getCurrentId: () => this.props.issueish.id,
+      refresh: this.refresh,
+      minimumIntervalPerId: 2 * 60 * 1000,
+    });
+    // auto-refresh disabled for now until pagination is handled
+    // this.refresher.start();
+  }
+
+  componentWillUnmount() {
+    this.refresher.destroy();
+  }
+
+  renderIssueBody(issueish, childProps) {
+    return (
+      <Fragment>
+        <GithubDotcomMarkdown
+          html={issueish.bodyHTML || '<em>No description provided.</em>'}
+          switchToIssueish={this.props.switchToIssueish}
+        />
+        <EmojiReactionsView reactionGroups={issueish.reactionGroups} />
+        <IssueTimelineController
+          {...childProps}
+          switchToIssueish={this.props.switchToIssueish}
+        />
+      </Fragment>
+    );
+  }
+
+  render() {
+    const repo = this.props.repository;
+    const issueish = this.props.issueish;
+    const childProps = {
```

Comment:
I think some child components that are used by both issues and pull requests want to know which is which, and we're not splitting all the components.  So I'm going to leave this as is, but double check that it's truly required by these children.